src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#ifndef SRC_MESH_MFEMDATA_HPP_
+#define SRC_MESH_MFEMDATA_HPP_
+
+#include <set>
+#include <vector>
+
+#include "mfem.hpp"
+
+#include "axom/core.hpp"
+#include "redecomp/redecomp.hpp"
+
+#include "tribol/types.hpp"
+#include "tribol/common/Parameters.hpp"
+#include "tribol/mesh/MethodCouplingData.hpp"
+
+namespace tribol
+{
+
+/**
+ * @brief Facilitates transfer to submesh low-order refined mesh
+ *
+ * This class simplifies transferring 1) a variable (e.g. displacement) from a
+ * higher-order grid function on a coarse mesh to a low-order grid function on a
+ * fine mesh and 2) a variable in the dual space (e.g. force) from a low-order
+ * grid function on a fine mesh to a higher-order grid function on a coarse
+ * mesh. The coarse mesh is intended to live on a mfem::ParSubMesh level, but
+ * this is not required to use this class.
+ */
+class SubmeshLORTransfer
+{
+public:
+  /**
+   * @brief Construct a new SubmeshLORTransfer object
+   * 
+   * @param ho_submesh_fes Higher order finite element space on the coarse mesh
+   * @param lor_submesh Refined mesh
+   */
+  SubmeshLORTransfer(
+    mfem::ParFiniteElementSpace& ho_submesh_fes,
+    mfem::ParMesh& lor_submesh
+  );
+
+  /**
+   * @brief Transfers data from a higher-order grid function on a coarse mesh
+   *
+   * Data is transferred to the low-order grid function on a fine mesh that can
+   * be accessed using GetLORGridFn().
+   *
+   * @param ho_src Source higher-order grid function on the coarse mesh
+   */
+  void TransferToLORGridFn(
```

Comment:
Per comments in .cpp file, I think it is often clearer and more consistent (per future discussion) to indicate transfer from and to, e.g. TransferHOToLORGridFn(). That lexicon has helped me a lot.